dkist/utils/_model_to_graphviz.py

```diff
@@ -0,0 +1,181 @@
+"""
+This module vizualises compound models by rendering them as a graphviz graph.
+
+It is provided as a helper module which is useful for debugging complex compound models.
+
+It is not tested, incomplete and *should be used at your own risk
+"""
+
+from functools import singledispatch
+
+import pydot
+
+from astropy.modeling import CompoundModel, Model
+from astropy.modeling.mappings import Mapping
+
+
+@singledispatch
+def to_subgraph(model, inputs: tuple[str] = None, outputs: tuple[str] = None):
+    """
+    Parameters
+    ----------
+    model
+        The model instance
+    inputs
+        The names to use for the model inputs, must be the same length as
+        model.n_inputs.
+    outputs
+        The names to use for the model outputs, must be the same length as
+        model.n_outputs.
+    """
+    raise NotImplementedError("Nope")
+
+
+@to_subgraph.register(Model)
+def model_to_subgraph(model, inputs=None, outputs=None):
+    inputs = inputs or model.inputs
+    outputs = outputs or model.outputs
+
+    input_labels = inputs
+    output_labels = outputs
+
+    if isinstance(model, Mapping):
```

Comment:
This line makes me think we need another dispatch function registered for a `Mapping` argument. (It would just call this `Model` function).

Reply:
I think that would be hard to do because you need to just change `output_labels` not outputs, so you would have to refactor things more than I am willing to do for a debug script :wink: